### PR TITLE
Vectorize temperatures logits

### PR DESCRIPTION
**Pull request by rd4com**
Hello,
it need to run some benchmarks to confirm it (work in progress)

### CODE REVIEW

**Comment by PriNova** [comment]
With this change I got an improvement of about 1.8% from 530 tok/sec to 540 tok/sec.

**Comment by tairov** [comment]
@rd4com would you mind resolve conflicts ?

**Comment by tairov** [comment]
still can't merge.. need help? 😄 
image

**Comment by rd4com** [comment]
yes i am quite new to github, if you know what to do tell me?
or you can just incorporate the changes yourself into the repo no big deal :+1: 
i use github desktop
